apps/api/src/app/workflows-v2/usecases/delete-workflow/delete-workflow.usecase.ts

```diff
@@ -0,0 +1,63 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+
+import { EnvironmentRepository, NotificationTemplateRepository } from '@novu/dal';
+import { DeleteWorkflowCommand } from './delete-workflow.command';
+import { ControlValuesRepository, MessageTemplateRepository } from '@novu/dal/src';
+import { WorkflowNotFoundException } from '../../exceptions/workflow-not-found';
+
+@Injectable()
+export class DeleteWorkflowUseCase {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private environmentRepository: EnvironmentRepository,
+    private msgTemplateRepo: MessageTemplateRepository,
+    private controlValuesRepository: ControlValuesRepository
+  ) {}
+
+  async execute(command: DeleteWorkflowCommand): Promise<void> {
+    await this.validateEnvironment(command);
+    const workflow = await this.findById(command);
+    if (!workflow) {
+      throw new WorkflowNotFoundException(command.workflowId);
+    }
+    await this.deleteComponents(command, workflow);
+  }
+
+  private async deleteComponents(command: DeleteWorkflowCommand, workflow) {
+    await this.notificationTemplateRepository.delete(buildDeleteQuery(command));
+    await this.deleteMsgTemplatesIfRequired(workflow, command);
+    this.controlValuesRepository.de({});
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Is `de({})` a typo? It's not clear what action the `de` method is taking.‏

Reply:
I was working on rebasing issues, you reviewed it in the middle, it didn't compile was finished right after and pushed